### PR TITLE
Allow users to upgrade their iOS engines without removing their edits to the Xcode project

### PR DESCRIPTION
**Pull request by chinmaygarde**
cc @devoncarew

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/ios.dart [hunk 1]
```diff
@@ -59,6 +59,11 @@ class IOSCommand extends FlutterCommand {
     tempFile.writeAsBytesSync(archiveBytes);
 
     try {
+      // Remove the old generated project if one is present
+      runCheckedSync(['/bin/rm', '-rf', directory]);
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/ios.dart]: We should probably prefer `dart:io` functions where they exist, over shelling out. So, `new Directory(directory).deleteSync(recurse: true)`.

File: packages/flutter_tools/lib/src/commands/ios.dart [hunk 2]
```diff
@@ -59,6 +59,11 @@ class IOSCommand extends FlutterCommand {
...
+      // Create the directory so unzip can write to it
+      runCheckedSync(['/bin/mkdir', '-p', directory]);
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/ios.dart]: ditto here

File: packages/flutter_tools/lib/src/commands/ios.dart [hunk 3]
```diff
@@ -122,21 +163,31 @@ class IOSCommand extends FlutterCommand {
...
-      printError("Warning: You may have made manual changes to files in the 'ios' directory.");
+      printError("Could not inflate the Xcode project archive.");
       return 1;
     }
 
-    // Step 3: Populate the Local.xcconfig with project specific paths
+    // Step 3: The generated project should NOT be checked into the users
+    //         version control system. Be nice and write a gitignore for them if
+    //         one does not exist.
+    File generatedGitignore = new File(path.join(iosFilesPath, ".gitignore"));
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/ios.dart]: We might instead ignore `.gitignore/Generated/` in the top level git ignore file created in the `create` command. There's also a path there and in the `.analysis_options` file that should be updated w/ the new ios paths.

**Comment by devoncarew** [comment]
some late comments :)

**Comment by chinmaygarde** [comment]
Thanks. Will fix these in the morning.

**Pull request merged**